### PR TITLE
FlightIcon pipeline - FigmaApi updates

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

Figma will [soon retire the old OAuth endpoints for their Figma REST API](https://www.figma.com/developers/api#oauth_migration_guide) (February 24, 2025).

This PR accounts for the changes to the underlying [`figma-api` library](https://github.com/didoo/figma-api) (maintained by @didoo) and takes the opportunity to update also some related dependencies.

Notice:
- while officially still in beta, the version `2.0` of the `figma-api` has been tested locally and it works as expected
- in the process, I have tried to update also `figma-export` to the latest version, but the script compilation was throwing an error (related to `require` vs `import`) so I decided to skip this update.

### :hammer_and_wrench: Detailed description

In this PR I have:
- bumped `typescript` to `5.6.3` (latest) in both `flight-icons` and `tokens` pipelines (to keep them in sync)
- bumped `ts-node` to `10.9.2` (latest) in `flight-icons` pipeline
- bumped `@types/node` to `22.8.7` (latest) in both `flight-icons` and `tokens` pipelines (to keep them in sync)
- bumped `figma-api` to `2.0.1-beta` in `flight-icons` pipeline
- updated the `figma-api` endpoint calls in our code to use the new format

### :link: External links

Jira ticket: https://hashicorp.atlassian.net/browse/HDS-4103

***

### 👀 Component checklist

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/tokens/package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
   "devDependencies": {
     "@types/fs-extra": "^11.0.3",
     "@types/lodash": "^4.14.200",
-    "@types/node": "^20.8.9",
+    "@types/node": "^22.8.7",
```
Comment (didoo) [inline packages/tokens/package.json:27]: @alex-ju if you prefer to have the changes to the token's `package.json` file in a separate PR let me know, I can easily do it. It was done mainly to keep the two pipelines in sync.

Comment (alex-ju) [reply packages/tokens/package.json:27]: no, I'm ok with them in the same PR – keeping dependencies in sync sounds good

**Review by alex-ju — COMMENTED** [review]
I found some issues in my testing.

`tokens` seems ok 👍 

`flight-icons` not so much:
 - type checking fails (`yarn run typecheck`)
 - syncing (`yarn run sync`) fails with my local token (error below) resulting in removing all icons; this could be a 'me' issue

```
Removing "svg-original" output folder
Retrieving assets metadata via REST API
[AxiosError: Request failed with status code 403]
```

**Comment by didoo** [comment]
> I found some issues in my testing.
> * type checking fails (`yarn run typecheck`)
> * syncing (`yarn run sync`) fails with my local token (error below) resulting in removing all icons; this could be a 'me' issue

@alex-ju I'll see what I can do (they were there also before this PR)
screenshot_4402

> ```
> Removing "svg-original" output folder
> Retrieving assets metadata via REST API
> [AxiosError: Request failed with status code 403]
> ```

@alex-ju do you have a `.env` file in your local branch, with the personal token? (details here: https://github.com/hashicorp/design-system/blob/main/packages/flight-icons/CONTRIBUTING.md#figma-token)

**Comment by alex-ju** [comment]
> > I found some issues in my testing.
> > 
> > * type checking fails (`yarn run typecheck`)
> > * syncing (`yarn run sync`) fails with my local token (error below) resulting in removing all icons; this could be a 'me' issue
> 
> @alex-ju I'll see what I can do (they were there also before this PR) screenshot_4402 

it works well for me in `main` 🤔 

> @alex-ju do you have a `.env` file in your local branch, with the personal token? (details here: https://github.com/hashicorp/design-system/blob/main/packages/flight-icons/CONTRIBUTING.md#figma-token)

yes, I have a local token setup, tested with it before, but maybe something changed in my permissions (my Figma account often gets downgraded to view only); if you tested `sync` and works well then I'm happy to approve

**Review by alex-ju — APPROVED** [review]

**Comment by didoo** [comment]
Currently investigating why the [GH action failed](https://github.com/hashicorp/design-system/actions/runs/11708171246) (as in @alex-ju local env) but it works in my local environment 🤔

**Comment by didoo** [comment]
OK, [mystery solved](https://hashicorp.slack.com/archives/C025N5V4PFZ/p1730912969004059?thread_ts=1730912946.488799&cid=C025N5V4PFZ). [All good](https://github.com/hashicorp/design-system/actions/runs/11708496053/job/32610383430), we can merge.

**Pull request merged**